snapshots/overlay/overlay.go

```diff
@@ -287,41 +287,37 @@ func (o *snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, k
 	if err != nil {
 		return nil, err
 	}
+	isRollback := true
```

Comment:
nit: just call it `rollback`

Reply:
I don't know why not call rollback when commit is fail before.
Do you know the reason?
```
	if err = t.Commit(); err != nil {
		return nil, errors.Wrap(err, "commit failed")
	}
```

Reply:
Should we call rollback when the Commit  success?

Reply:
I just meant renaming `isRollback` to `rollback` :sweat_smile: 

I don't know why the commit failure wouldn't cause a rollback previously (cc @dmcgowan)

Reply:
`Commit` already does a rollback when there is a failure on commit, calling `Rollback` would be `redundant`. The functions are designed that either `Commit` or `Rollback` need to be called, never both.

Reply:
@mlaventure  Sorry for my English.  shame. The pr has been merged. I will modify it in the other pr later.
@dmcgowan  Thanks

Reply:
@dmcgowan Seem like there are many place calling both commit and rollback.
for example:  naive.go
```
func (o *snapshotter) Remove(ctx context.Context, key string) (err error) {
	ctx, t, err := o.ms.TransactionContext(ctx, true)
	if err != nil {
		return err
	}
	defer func() {
		if err != nil && t != nil {
			if rerr := t.Rollback(); rerr != nil {
				log.G(ctx).WithError(rerr).Warn("failed to rollback transaction")
			}
		}
	}()

	id, _, err := storage.Remove(ctx, key)
	if err != nil {
		return errors.Wrap(err, "failed to remove")
	}

	path := o.getSnapshotDir(id)
	renamed := filepath.Join(o.root, "snapshots", "rm-"+id)
	if err := os.Rename(path, renamed); err != nil {
		if !os.IsNotExist(err) {
			return errors.Wrap(err, "failed to rename")
		}
		renamed = ""
	}

	err = t.Commit()
	t = nil
	if err != nil {
		if renamed != "" {
			if err1 := os.Rename(renamed, path); err1 != nil {
				// May cause inconsistent data on disk
				log.G(ctx).WithError(err1).WithField("path", renamed).Errorf("Failed to rename after failed commit")
			}
		}
		return errors.Wrap(err, "failed to commit")
	}
	if renamed != "" {
		if err := os.RemoveAll(renamed); err != nil {
			// Must be cleaned up, any "rm-*" could be removed if no active transactions
			log.G(ctx).WithError(err).WithField("path", renamed).Warnf("Failed to remove root filesystem")
		}
	}

	return nil
}
```

Reply:
@dmcgowan @stevvooe @crosbymichael 
I think we can refactor it.
e.g.
type Handle func(ctx context.Context) bool
func (ms *MetaStore) HandleTransaction(ctx context.Context, writable bool, f Handle) error {
	ctx, t, err := ms.TransactionContext(ctx, writable)
	if err != nil {
		return err
	}
	if f(ctx) {
		if writable {
			err = t.Commit()
		}
	}else{
		err = t.Rollback()
	}
	return err
}

HandleTransaction(ctx, writable, func(ctx context.Context) bool {
	user xxxHandle
})

Reply:
@dmcgowan @stevvooe @crosbymichael   WDYT?